crates/jiter/src/number_decoder.rs

```diff
@@ -384,8 +384,8 @@ impl AbstractNumberDecoder for NumberRange {
                         let end = consume_exponential(data, index)?;
                         Ok((start..end, end))
                     }
-                    Some(_) => return json_err!(InvalidNumber, index),
-                    None => return Ok((start..index, index)),
+                    Some(digit) if digit.is_ascii_digit() => json_err!(InvalidNumber, index),
```

Comment:
Doesn't this mean that things like `0a` are now unintentionally valid?

Reply:
no, the next step takes care, this matches the normal number parser.